st2api/tests/unit/controllers/v1/test_traces.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2tests.fixturesloader import FixturesLoader
+from tests import FunctionalTest
+
+FIXTURES_PACK = 'traces'
+
+TEST_MODELS = {
+    'traces': [
+        'trace_empty.yaml',
+        'trace_one_each.yaml',
+        'trace_multiple_components.yaml'
+    ]
+}
+
+
+class TestTraces(FunctionalTest):
+
+    models = None
+    trace1 = None
+    trace2 = None
+    trace3 = None
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestTraces, cls).setUpClass()
+        cls.models = FixturesLoader().save_fixtures_to_db(fixtures_pack=FIXTURES_PACK,
+                                                          fixtures_dict=TEST_MODELS)
+        cls.trace1 = cls.models['traces']['trace_empty.yaml']
+        cls.trace2 = cls.models['traces']['trace_one_each.yaml']
+        cls.trace3 = cls.models['traces']['trace_multiple_components.yaml']
+
+    def test_get_all(self):
+        resp = self.app.get('/v1/traces')
+        self.assertEqual(resp.status_int, 200)
+        self.assertEqual(len(resp.json), 3, '/v1/traces did not return all traces.')
+
+        retrieved_trace_ids = [trace['trace_id'] for trace in resp.json]
+
+        self.assertEqual(retrieved_trace_ids,
+                         [self.trace1.trace_id, self.trace2.trace_id, self.trace3.trace_id],
+                         'Incorrect traces retrieved.')
+
+    def test_query_by_trace_id(self):
+        resp = self.app.get('/v1/traces/?trace_id=test-trace-1')
+        self.assertEqual(resp.status_int, 200)
+        self.assertEqual(len(resp.json), 1, '/v1/traces?trace_id=x did not return correct trace.')
+
+        self.assertEqual(resp.json[0]['trace_id'], self.trace1['trace_id'],
+                         'Correct trace not returned.')
+
+    def test_query_by_action_execution(self):
+        resp = self.app.get('/v1/traces/?action_execution=action_execution_2')
```

Comment:
I am really not sure what this means? Are you looking for trace where execution id is foo? If so, let's use object id in the example so it's less confusing.